cli/crates/graphql-introspection/src/subgraph.rs

```diff
@@ -0,0 +1,72 @@
+use std::collections::HashMap;
+
+use reqwest::header::USER_AGENT;
+use url::Url;
+
+use crate::Error;
+
+#[derive(Debug, serde::Deserialize)]
+struct Service {
+    sdl: String,
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct Data {
+    #[serde(rename = "_service")]
+    service: Service,
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct Response {
+    data: Option<Data>,
```

Comment:
Would we be interested in `errors` here too? The status isn't always the indicator for GraphQL APIs.

Reply:
What we figure out here is if data is `None`, we have errors and that's an indicator to do the standard introspection. I'm not super interested about what the error data contains...